move2kube/exercise-3-customizations/exercise-3a-overriding-built-in-transformers/README.md

```diff
@@ -1 +1,130 @@
+
+    ```console
+    $ ls myproject/source/frontend
+    ```
+
+    As you notice, there are no scripts named `start-nodejs.sh` in the frontend service directory.
+    
+    ```console
+    Dockerfile        __mocks__         jest.config.js    package-lock.json src               test-setup.js     webpack.dev.js    LICENSE           dist              manifest.yml      package.json      stories           tsconfig.json     webpack.prod.js   README.md         dr-surge.js       nodemon.json      server.js         stylePaths.js     webpack.common.js
+    ```
+
+3. Check the generated Dockerfile for frontend service of the `enterprise-app`:
+
+    ```console
+    $ cat myproject/source/frontend/Dockerfile
+    ```
+
+    If we notice the Dockerfile generated for the frontend app, it uses registry.access.redhat.com/ubi8/nodejs-12 as base image.
+    
+    ```console
+    FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
+    RUN microdnf update && microdnf install wget xz tar && microdnf clean all && \
+        wget https://nodejs.org/dist/v14.20.0/node-v14.20.0-linux-x64.tar.xz && \
+        tar -xJf node-v14.20.0-linux-x64.tar.xz && \
+        mv node-v14.20.0-linux-x64 /node && \
+        rm -f node-v14.20.0-linux-x64.tar.xz
+    ENV PATH="$PATH:/node/bin"
+    COPY . .
+    RUN npm install
+    RUN npm run build
+    USER root
+    RUN chown -R 1001:0 /opt/app-root/src/.npm
+    RUN chmod -R 775 /opt/app-root/src/.npm
+    USER 1001
+    EXPOSE 8080
+    CMD npm run start
+    ```
+
+4. List the deployment artifacts generated by Move2Kube:
+
+    ```console
+    ls myproject/deploy
+    ```
+
+    If you notice, the Kubernetes yamls are generated in myproject/deploy/yamls directory.
+    
+    ```console
+    cicd                  compose               knative               knative-parameterized yamls                 yamls-parameterized
+    ```
+
+5. Lets customize the generated output. We will use a [custom configured version](https://github.com/konveyor/move2kube-transformers/tree/main/custom-dockerfile-change-built-in-behavior) of the [nodejs built-in transformer](https://github.com/konveyor/move2kube/tree/main/assets/built-in/transformers/dockerfilegenerator/nodejs) and [kubernetes built-in transformer](https://github.com/konveyor/move2kube/tree/main/assets/built-in/transformers/kubernetes/kubernetes) to achieve this. We already have these in this exercise in the `customizations` folder.
+
+6. Run Move2Kube transform with the customizations, specifying the customization using the -c flag.
+
+    ```console
+    $ move2kube transform -s ../../source/enterprise-app/ -c customizations/ --qa-skip --overwrite
+    ```
+
+7. Check the modified Dockerfile for frontend service of the `enterprise-app`:
+
+    ```console
+    $ cat myproject/source/frontend/Dockerfile
+    ```
+
+    Observe the base image is changed from `registry.access.redhat.com/ubi8/nodejs-12` to `quay.io/konveyor/nodejs-12` in the generated nodejs dockerfile.
```

Comment:
Please check the base image and nodejs version in the Dockerfile generated using latest Move2Kube version without custom transformer. If required, please update the nodejs version in the custom Dockerfile template to match the version in the original Dockerfile.

Reply:
Fixed!